lib/kubeclient/common.rb

```diff
@@ -78,11 +92,48 @@ def handle_exception
       raise KubeException.new(e.http_code, err_message, e.response)
     end
 
+    def discover
+      @entities = {}
+      handle_exception do
+        response = rest_client.get(@headers)
+        result = JSON.parse(response)
+        result['resources'].each do |resource|
+          # mitigation for https://github.com/kubernetes/kubernetes/issues/22413:
+          next if resource['name'].include?('/') || resource['kind'] == 'ReplicationControllerDummy'
+          entity = ClientMixin.parse_definition(resource['kind'], resource['name'])
+          @entities[entity.method_names[0]] = entity if entity
+        end
+        define_entity_methods
+        @discovered = true
+      end
+    end
+
+    def self.parse_definition(kind, name)
+      # "name": "componentstatuses", networkpolicies, endpoints
+      # "kind": "ComponentStatus" NetworkPolicy, Endpoints
+      # maintain pre group api compatibility for endpoints.
+      # See: https://github.com/kubernetes/kubernetes/issues/8115
+      kind = 'Endpoint' if kind == 'Endpoints'
```

Comment:
Forgot to remove the exception for ReplicationControllerDummy after implementing a general solution for non verbs.